sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java

```diff
@@ -127,12 +132,22 @@ public synchronized void start() {
      * {@codesnippet com.azure.messaging.eventhubs.eventprocessorclient.startstop}
      */
     public synchronized void stop() {
-        if (!started.compareAndSet(true, false)) {
+        if (!isRunning.compareAndSet(true, false)) {
             logger.info("Event processor has already stopped");
             return;
         }
         runner.get().dispose();
         scheduler.get().dispose();
         this.partitionPumpManager.stopAllPartitionPumps();
     }
+
+    /**
+     * Returns {@code true} if the event processor is running. If the event processor is already running, calling
+     * {@link #start()} has no effect.
+     *
+     * @return {@code true} if the event processor is running.
+     */
+    public synchronized boolean isRunning() {
```

Comment:
Does this need to be synchronised? isRunning.get() is atomic

Reply:
Yes, `isRunning` may have been updated but the start or stop operation may not have been fully completed.

Reply:
Ahh, I didn't realise that synchronized is across the entire object. 👍